config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
+var EnvironmentError = errors.New("environment variable not found")
+
+var regex *regexp.Regexp
+
+func init() { regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`) }
+
+func replaceEnvVar(in string) (string, error) {
+
+	// TODO(ear7h): FindAllString
+	varName := regex.FindString(in)
+	// no env var
+	if len(varName) == 0 {
+		return in, nil
+	}
+
+	// trim the leading '${' and trailing '}'
+	varName = varName[2:len(varName)-1]
+
+	// get env var
+	envVar := os.Getenv(varName)
+	if envVar == "" {
+		return "", EnvironmentError
+	}
+
+	return regex.ReplaceAllString(in, envVar), nil
+}
+
+type Bool bool
+func BoolPtr(v Bool) *Bool {
+	return &v
+}
+
+func (t *Bool) UnmarshalTOML(v interface{}) error {
+	var boolVal bool
+	var err error
+
+	switch val := v.(type) {
+	case string:
+		val, err = replaceEnvVar(val)
+		if err != nil {
+			break
+		}
+
+		boolVal, err = strconv.ParseBool(val)
+	case bool:
+		boolVal = val
+	default:
+		err = TypeError
+	}
+
+	if err != nil {
+		return err
+	}
+
+	*t = Bool(boolVal)
+	return nil
+}
+
+type String string
+func StringPtr(v String) *String {
+	return &v
+}
+
+func (t *String) UnmarshalTOML(v interface{}) error {
+	var stringVal string
+	var err error
+
+	switch val := v.(type) {
+	case string:
+		stringVal, err = replaceEnvVar(val)
+		if err != nil {
+			break
+		}
+	default:
+		err = TypeError
+	}
+
+	if err != nil {
+		return err
+	}
+
+	*t = String(stringVal)
+	return nil
+}
+
+type Uint uint
+func UintPtr(v Uint) *Uint {
+	return &v
+}
+
+func (t *Uint) UnmarshalTOML(v interface{}) error {
+	var uintVal uint64
+	var err error
+
+	switch val := v.(type) {
+	case string:
+		val, err = replaceEnvVar(val)
+		if err != nil {
+			break
+		}
+		uintVal, err = strconv.ParseUint(val, 10, 64)
+	case int:
```

Comment:
Our config format currently only needs `uint` support, but I don't think we should be conflating `int` and `uint` here. We should support `int` as a separate type if necessary.